lib/kadai-core/src/main/java/io/kadai/user/internal/UserServiceImpl.java

```diff
@@ -161,17 +161,20 @@ public User updateUser(User userToUpdate)
   @Override
   public void deleteUser(String id)
       throws UserNotFoundException, InvalidArgumentException, NotAuthorizedException {
+
     internalKadaiEngine.getEngine().checkRoleMembership(KadaiRole.BUSINESS_ADMIN, KadaiRole.ADMIN);
-    getUser(id);
+
+    User user = getUser(id);
+    String userId = user.getId();
```

Comment:
Because id comes directly from the REST layer without any sort of validation. Since `getUser(id)` is call to validate that the user exists, it is better to use the id from the retrieved object, than to directly use something that comes from the REST layer. And since `getUser(id)` was already called, there are no additional instructions executed.